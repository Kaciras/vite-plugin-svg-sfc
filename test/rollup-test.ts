import { expect, it } from "vitest";
import { OutputAsset, OutputChunk, rollup } from "rollup";
import { extractSFCPlugin, resolveFixture } from "./test-helper.ts";
import svgSfc from "../index.ts";

it("should work with rollup", async () => {
	const build = await rollup({
		input: resolveFixture("instruction.svg?sfc"),
		plugins: [svgSfc(), extractSFCPlugin],
	});
	const result = await build.generate({});

	const [chunk, asset] = result.output;
	expect((asset as OutputAsset).source).toMatchSnapshot();
	expect((chunk as OutputChunk).facadeModuleId).toMatch(/\.svg\.vue\?sfc/);
});
