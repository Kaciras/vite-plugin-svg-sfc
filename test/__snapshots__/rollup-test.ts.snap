// Vitest Snapshot v1

exports[`should work with rollup 1`] = `"<template><svg version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" fill=\\"currentColor\\" height=\\"1em\\" width=\\"1em\\"><path d=\\"M0 0h24v24H0zm0 0h24v24H0zm0 0h24v24H0zm0 0h24v24H0z\\"/></svg></template>"`;
